backend/PrintAsmaux.ml

```diff
@@ -0,0 +1,135 @@
+(*              The Compcert verified compiler                         *)
+(*                                                                     *)
+(*          Xavier Leroy, INRIA Paris-Rocquencourt                     *)
+(*          Bernhard Schommer, AbsInt Angewandte Informatik GmbH       *)
+(*                                                                     *)
+(*  Copyright Institut National de Recherche en Informatique et en     *)
+(*  Automatique.  All rights reserved.  This file is distributed       *)
+(*  under the terms of the INRIA Non-Commercial License Agreement.     *)
+(*                                                                     *)
+(* *********************************************************************)
+
+open AST
+open Asm
+open Camlcoq
+open Datatypes
+open Printf
+open Sections
+
+(** Auxiliary functions for printing of asm *)
+
+module type TARGET =
+    sig
+      val print_prologue: out_channel -> unit
+      val print_epilogue: out_channel -> unit
+      val print_align: out_channel -> int -> unit
+      val print_comm_symb:  out_channel -> Z.t -> P.t -> int -> unit
+      val print_var_info: bool
+      val print_fun_info: bool
+      val print_init: out_channel -> init_data -> unit
+      val reset_constants: unit -> unit
+      val get_section_names: P.t -> section_name * section_name * section_name
+      val print_file_line: out_channel -> string -> int -> unit
+      val print_optional_fun_info: out_channel -> unit
+      val cfi_startproc: out_channel -> unit
+      val print_instructions: out_channel -> coq_function -> unit
+      val cfi_endproc: out_channel -> unit
+      val emit_constants: out_channel -> section_name -> unit
+      val print_jumptable: out_channel -> section_name -> unit
+      val section: out_channel -> section_name -> unit
+      val name_of_section: section_name -> string
+      val comment: string
+    end
+
+(* On-the-fly label renaming *)
+
+let next_label = ref 100
+
+let new_label () =
+  let lbl = !next_label in incr next_label; lbl
+
+let current_function_labels = (Hashtbl.create 39 : (label, int) Hashtbl.t)
+
+let transl_label lbl =
+  try
+    Hashtbl.find current_function_labels lbl
+  with Not_found ->
+    let lbl' = new_label() in
+    Hashtbl.add current_function_labels lbl lbl';
+    lbl'
+
+let label oc lbl =
+  fprintf oc ".L%d" lbl
+
+(* List of literals and jumptables used in the code *)
+
+let float64_literals : (int * int64) list ref = ref []
+let float32_literals : (int * int32) list ref = ref []
+let jumptables : (int * label list) list ref = ref []
+
+let reset_constants () =
+  float64_literals := [];
+  float32_literals := [];
+  jumptables := []
+
+(* Variables used for the handling of varargs *)
+
+let current_function_stacksize = ref 0l
+let current_function_sig =
+  ref { sig_args = []; sig_res = None; sig_cc = cc_default }
+
+(* Functions for printing of symbol names *)
+let symbol oc symb =
```

Comment:
Again, these definitions are good for ELF but not for every target.  I can understand you want to avoid duplicating them in every ELF Target, but then what about calling them "elf_symbol" and "elf_symbol_offset" ?